### DISCUSSION TITLE
Python CGI endless error

### DISCUSSION THREAD

**Discussion by LEFT-BEE** [community/community#21491] [Other Feature Feedback, Questions, & Ideas]
<pre><code class="lang-auto">127.0.0.1 - - [11/Jul/2020 20:03:02] command: C:\Users\2포대\AppData\Local\Programs\Python\Python37\python.exe -u C:\Users\2포대\cgi-bin\chatbot.py ""
127.0.0.1 - - [11/Jul/2020 20:03:03] b'Traceback (most recent call last):\r\n  File "C:\\Users\\2\xc6\xf7\xb4\xeb\\cgi-bin\\chatbot.py", line 5, in &lt;module&gt;\r\n    from botengine import make_reply\r\n  File "C:\\Users\\2\xc6\xf7\xb4\xeb\\cgi-bin\\botengine.py", line 8, in &lt;module&gt;\r\n    okt = Okt()\r\n  File "C:\\Users\\2\xc6\xf7\xb4\xeb\\AppData\\Local\\Programs\\Python\\Python37\\lib\\site-packages\\konlpy\\tag\\_okt.py", line 95, in __init__\r\n    self.jki = OktInterfaceJavaClass()\r\n  File "C:\\Users\\2\xc6\xf7\xb4\xeb\\AppData\\Local\\Programs\\Python\\Python37\\lib\\site-packages\\jpype\\_jpackage.py", line 123, in __call__\r\n    raise TypeError("Package {0} is not Callable".format(self._name))\r\nTypeError: Package &lt;Java package kr.lucypark.okt.OktInterface._name&gt; is not Callable\r\n'
127.0.0.1 - - [11/Jul/2020 20:03:03] CGI script exit status 0x1
</code></pre>
<p>this is error code</p>
<pre><code class="lang-auto">
#!/usr/bin/env python3
#html에서 입력할시 실행된다.
import cgi#웹 관련 라이브러리
from botengine import make_reply

# 입력 양식의 글자 추출하기 --- (※1)
form = cgi.FieldStorage() # 웹에서 입력값을 받아 form 의 내용을
# form 이라는 변수에 저장함

# 메인 처리 --- (※2)
def main():
    m = form.getvalue("m", default="")
 
    show_form()#웹브라우저의 HTML출력

# 사용자의 입력에 응답하기 --- (※3)
def api_say():
    print("Content-Type: text/plain; charset=utf-8")
    print("")
    txt = form.getvalue("txt", default="")
    if txt == "": return
    res = make_reply(txt)#마르코프체인문장만들기
    print(res)

# 입력 양식 출력하기 --- (※4)
def show_form():
    print("Content-Type: text/html; charset=utf-8")
    print("")
    print("""
    &lt;html&gt;&lt;meta charset="utf-8"&gt;&lt;body&gt;
    &lt;script src="https://code.jquery.com/jquery-3.1.1.min.js"&gt;&lt;/script&gt;
    &lt;style&gt;
        h1   { background-color: #ffe0e0; }
        div  { padding:10px; }
        span { border-radius: 10px; background-color: #ffe0e0; padding:8px; }
        .bot { text-align: left; }
        .usr { text-align: right; }
    &lt;/style&gt;
    &lt;h1&gt;대화하기&lt;/h1&gt;
    &lt;div id="chat"&gt;&lt;/div&gt;
    &lt;div class='usr'&gt;&lt;input id="txt" size="40"&gt;
    &lt;button onclick="say()"&gt;전송&lt;/button&gt;&lt;/div&gt;
    &lt;script&gt;
    var url = "./chatbot.py";
    function say() {
      var txt = $('#txt').val();
      $.get(url, {"m":"say","txt":txt},
        function(res) {
          var html = "&lt;div class='usr'&gt;&lt;span&gt;" + esc(txt) +
            "&lt;/span&gt;: 나&lt;/div&gt;&lt;div class='bot'&gt; 봇:&lt;span&gt;" + 
            esc(res) + "&lt;/span&gt;&lt;/div&gt;";
          $('#chat').html($('#chat').html()+html);
          $('#txt').val('').focus();
        });
    }
    function esc(s) {
        return s.replace('&amp;', '&amp;amp;').replace('&lt;','&amp;lt;')
                .replace('&gt;', '&amp;gt;');
    }
    &lt;/script&gt;&lt;/body&gt;&lt;/html&gt;
    """)

main()

</code></pre>
<p>this is my code</p>
<p>i don’t know web server well  but i guess problem is cgi</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/8b84320282dcc26526a14e8298533149a73fefe8.png" data-download-href="/uploads/short-url/jUdAVAbMFPNco2plt2i7glKv4Bq.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/65720894/181065010-9cf1953e-70a9-4bfd-97fe-81f0c2c6cc27.png" alt="image" data-base62-sha1="jUdAVAbMFPNco2plt2i7glKv4Bq" width="607" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/8b84320282dcc26526a14e8298533149a73fefe8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">859×707 20.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
and this is web

**Comment by airtower-luna** [comment, marked as answer]
<p>Take a look at that traceback in the server log, decoding the Python <code>bytes</code> object makes it a bit more readable:</p>
<pre><code class="lang-auto">&gt;&gt;&gt; print(b.decode(errors='replace'))
Traceback (most recent call last):
  File "C:\Users\2����\cgi-bin\chatbot.py", line 5, in &lt;module&gt;
    from botengine import make_reply
  File "C:\Users\2����\cgi-bin\botengine.py", line 8, in &lt;module&gt;
    okt = Okt()
  File "C:\Users\2����\AppData\Local\Programs\Python\Python37\lib\site-packages\konlpy\tag\_okt.py", line 95, in __init__
    self.jki = OktInterfaceJavaClass()
  File "C:\Users\2����\AppData\Local\Programs\Python\Python37\lib\site-packages\jpype\_jpackage.py", line 123, in __call__
    raise TypeError("Package {0} is not Callable".format(self._name))
TypeError: Package &lt;Java package kr.lucypark.okt.OktInterface._name&gt; is not Callable
</code></pre>
<p>Looks like importing the <code>botengine</code> package fails. Not sure what’s going on with the Java stuff, maybe that module requires Jython and certain Java packages or something?</p>
<p>As a side note, that traceback isn’t valid <code>utf-8</code> (that’s why I used <code>errors='replace'</code> for decoding), you might want to fix whatever locale setting is responsible for that.</p>
